src/annotator/util/extract-annotation-query.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+/**
+ * Extracts an annotation selection or default filter from a url.
+ *
+ * @param {string} url - The URL which may contain a '#annotations:<ID>'
+ *        fragment.
+ * @return {Object} - An object with either an annotation ID or a filter string.
+ */
+function extractAnnotationQuery(url) {
+  var filter = {};
+  try {
+    // Annotation IDs are url-safe-base64 identifiers
+    // See https://tools.ietf.org/html/rfc4648#page-7
+    var annotFragmentMatch = url.match(/#annotations:([A-Za-z0-9_-]+)$/);
+    var queryFragmentMatch = url.match(/#annotations:(query|q):(.+)$/i);
+    if (queryFragmentMatch) {
+      filter.query = queryFragmentMatch[2];
+    } else if (annotFragmentMatch) {
+      filter.annotations = annotFragmentMatch[1];
+    } else {
+      filter = null;
+    }
+  } catch (err) {
+    filter = null;
```

Comment:
After trying to break the code without the catch, a URI error can occur if someone produces an annotation query with invalid identifier (e.g. annotations: %%%%%%%%%).  It should through a filter = null in this case.  I will target that error for now and see if I can think of any other ways to break it.